### PR TITLE
feat: initial kubectl layer construct

### PR DESCRIPTION
**Pull request by kaizencc**
This is nearly a direct copy from https://github.com/aws/aws-cdk/tree/main/packages/%40aws-cdk/lambda-layer-kubectl with some changes to make it v2-compatible and projen-compatible.

Note: this exposes a `kubeCtlLayer` construct, not the final form `kubeCtlAsset`. I figure it is better to start here and then update this repo in subsequent PRs. However, that means there are some inconsistencies with what I say this library exposes -- since this is not meant for active use right now, I feel like that is ok and will be corrected in a future PR.

### CODE REVIEW

File: .projenrc.js [hunk 1]
```diff
@@ -1,15 +1,39 @@
...
+const { awscdk, JsonPatch } = require('projen');
 const project = new awscdk.AwsCdkConstructLibrary({
-  author: 'Kaizen Conroy',
-  authorAddress: 'zen.conroy@gmail.com',
-  cdkVersion: '2.1.0',
+  author: 'Amazon Web Services',
+  cdkVersion: '2.0.0',
   defaultReleaseBranch: 'main',
-  name: 'awscdk-asset-kubectl',
+  name: '@aws-cdk/asset-kubectl-v1-20',
```
Comment (kaizencc) [inline .projenrc.js:6]: Naming suggestions? `@aws-cdk/asset-kubectl-v1.20` does not work. I don't know if people care about the minor version of kubectl here, maybe just `@aws-cdk/asset-kubectl-v1` is sufficient?

Comment (madeline-k) [reply .projenrc.js:6]: people do care about the minor version. Each minor version bump in kubernetes can have breaking changes. How about we borrow the convention from cdk8s-plus https://www.npmjs.com/package/cdk8s-plus-21

`@aws-cdk/asset-kubectl-21`

Comment (kaizencc) [reply .projenrc.js:6]: Ok. Looks like this should be `@aws-cdk/asset-kubectil-v20` then.

**Comment by kaizencc** [comment]
> Is there an integration test to bring over from aws-cdk? Otherwise, looks good

No integ test in this module

**Review by madeline-k — APPROVED** [review]

**Pull request merged**